r/tests/testthat/test-Array.R

```diff
@@ -459,6 +459,53 @@ test_that("Array$create() can handle data frame with custom struct type (not inf
   expect_error(Array$create(df, type = type), regexp = "Cannot convert R object to string array")
 })
 
+test_that("Array$create() handles vector -> list arrays (ARROW-7662)", {
+  verify <- function(v, type) {
```

Comment:
Can you give this function a more expressive name, like `expect_list_array` or something? I found it confusing at first glance to have the same function name here and in the next block but they do different things.